File: src/i_interoperability/interoperability.java
package i_interoperability;

import h_classes.KotlinUserDataClass;
import h_classes.KotlinUserRegularClass;

public class interoperability {
    public static void main(String[] args) {
        var kotlinUserDataClassInstance = new KotlinUserDataClass("Alice", "+49-012345678910");

        System.out.println(kotlinUserDataClassInstance.getName());
        System.out.println(kotlinUserDataClassInstance.getPhoneNumber());

        var kotlinUserRegularClass = new KotlinUserRegularClass("Bob", "+49-9876543210");
        kotlinUserRegularClass.printUser();
    }
}

File: src/i_interoperability/interoperability.kt
package i_interoperability

import h_classes.JavaUserRecord
import h_classes.JavaUserRegularClass

fun main(args: Array<String>) {
    val javaUserRecord = JavaUserRecord("Alice", "+49-012345678910")
    println(javaUserRecord.name)
    println(javaUserRecord.phoneNumber)

    val javaUserRegularClass = JavaUserRegularClass("Bob", "+49-012345678911")
    javaUserRegularClass.printUser()
}
